File: RAI/Clase16/Java/AbstractClasses/Gato.java
//package RAI.Clase16.Java.AbstractClasses;

public class Gato extends Animal{

    private String raza;

    public Gato(String nombre, String especie, String raza){
        super(nombre, especie);
        this.raza = this.raza;
    }

    public void haceSonido(){
        System.out.println(this.nombre + " hace Miauu...");
    }
    public void ronrronear(){
        System.out.println(this.nombre + " hace purrr...");
    }
    
}

File: RAI/Clase16/Python/AbstractClasses/Gato.py
from Animal import Animal

class Gato(Animal):

    def __init__(self, nombre:str, especie:str, raza:str):
        self.nombre = nombre
        self.especie = especie
        self.raza = raza

    def hace_sonido(self):
        print(self.nombre + " hace Miau...")